File: backend/src/models/employee.py
from sqlmodel import SQLModel, Field, Relationship
from typing import Optional, List
from pydantic import EmailStr

class EmployeeBase(SQLModel):
    name: str = Field(max_length=255)
    email: EmailStr = Field(max_length=100)
    code: str = Field(max_length=45)
    lastname: str = Field(max_length=100)

class Employee(EmployeeBase, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    enterprise_id: Optional[int] = Field(default=None, foreign_key="enterprise.id")
    role_id: Optional[int] = Field(default=None, foreign_key="role.id")
    is_active: bool = Field(default=True)
    hashed_password: str = Field(max_length=255)
    
    enterprise: Optional["Enterprise"] = Relationship(back_populates="employees")
    role: Optional["Role"] = Relationship(back_populates="employees")

class EmployeeCreate(EmployeeBase):
    password: str
    enterprise_id: int
    role_id: int

class EmployeeRead(EmployeeBase):
    id: int
    enterprise_id: int
    role_id: int
    is_active: bool

class EmployeeUpdate(SQLModel):
    name: Optional[str] = None
    email: Optional[EmailStr] = None
    code: Optional[str] = None
    lastname: Optional[str] = None
    password: Optional[str] = None
    is_active: Optional[bool] = None
    role_id: Optional[int] = None

File: frontend/services/api/employee.ts
import api from '../config/api';

export interface Employee {
    id: number;
    name: string;
    email: string;
    code: string;
    lastname: string;
    telephone: string;
    enterprise_id: number;
    role_id: number;
    is_active: boolean;
}

export interface EmployeeCreate {
    name: string;
    email: string;
    code: string;
    lastname: string;
    telephone: string;
    password: string;
    enterprise_id: number;
    role_id: number;
}

export interface EmployeeUpdate {
    name?: string;
    email?: string;
    code?: string;
    lastname?: string;
    telephone?: string;
    password?: string;
    is_active?: boolean;
    role_id?: number;
}

export const employeeService = {
    async getAll(skip: number = 0, limit: number = 100) {
        const response = await api.get(`/employees?skip=${skip}&limit=${limit}`);
        return response.data;
    },

    async getById(id: number) {
        const response = await api.get(`/employees/${id}`);
        return response.data;
    },

    async create(data: EmployeeCreate) {
        const response = await api.post('/employees', data);
        return response.data;
    },

    async update(id: number, data: EmployeeUpdate) {
        const response = await api.put(`/employees/${id}`, data);
        return response.data;
    },

    async delete(id: number) {
        const response = await api.delete(`/employees/${id}`);
        return response.data;
    },

    async getMe() {
        const response = await api.get('/employees/me');
        return response.data;
    }
};